src/extension.ts

```diff
@@ -2828,10 +2835,8 @@ function is_valid_minimize_to_tray(meta_win: Meta.Window, ext: Ext) {
     let valid_min_to_tray = false;
     switch (meta_win.window_type) {
         case Meta.WindowType.NORMAL:
-        case Meta.WindowType.DIALOG:
-        case Meta.WindowType.UTILITY:
-        case Meta.WindowType.MODAL_DIALOG:
-            valid_min_to_tray = true;
+        case Meta.WindowType.UTILITY: // Gimp (Non-Single Window Mode)
+            valid_min_to_tray = !meta_win.is_override_redirect();
```

Comment:
Since this is a critically important piece, should it have a small comment to explain the reason for removing OR windows? Just to be sure future readings of the code won't be confused about this unusual function.

By the way, here are the questionable commits upstream, which may not be merged. You can see how much they had to change to start tracking OR windows in these patches:

- https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2029/diffs?commit_id=21e1aa3f36ab75f964ea68621bdde7d5466235b8
- https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2029/diffs?commit_id=4abde39369f762535d9e5330a4f5d8c2912e0268
- https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2029/diffs?commit_id=0c68e07ee4855a5dad2f44d4ce56eef6bc88f589

The 2nd and 3rd commit messages are the most interesting, because they explain why GNOME doesn't / didn't track OR windows. Basically that "there's no good reason" to track those windows. The 3rd commit explains that they had to hack some "actor system" to make GNOME Shell track OR windows, and they aren't happy about it upstream (talking about a rewrite or rejection of the OR patches).

So if we sum this up without referring to GNOME crashes (since we don't know yet if they will or won't merge the OR patches), we could comment this something like `// Don't track OR-windows since those are never allowed to be window managed.`

Reply:
Sure - I'll add that comment. FWIW, I am hesitant to add more changes to this PR by reading the upstream MR. Trying to minimize around users not experiencing `app is null` which I think is addressed already by checks for the minimal amount of window types and the override redirect

Reply:
@jmmaranan Yep exactly, I am of the view that we should fix this around the current GNOME behavior, not the potential extended tracking merge. And yep, all of the fixes we are talking about will fix the _current_ behavior! :)

Reply:
Again, thanks a lot for the comments and information. Let's probably wait for somebody from the System76 QA group before we move further. I believe additional changes might be required if anything has been broken implicitly.

Reply:
@jmmaranan Awesome. I will start testing your patches if they are ready. Did you want to rewrite anything else in here with the UTILITY "if not skip taskbar exclude" stuff, or is everything here ready for me to start testing stability?

Reply:
Nope - it is `all in` at this time and need to move to other things. :) and that would be awesome to get help on testing too

Reply:
@jmmaranan Awesome, you've done a really great job! I'll build it and start testing. Clean GNOME 41.1, patched Pop Shell. I'll be doing my usual workflow and also testing things like GIMP multi window mode. The code looks good so I think all will work as expected. :)

Ping @mmstick it's ready for System76 QA testing.